### PR TITLE
Skip compaction on non-partitioned tables

### PR DESCRIPTION
**Pull request by teamurko**
## Summary
Currently, there is no reason to run data compaction on non-partitioned tables since most of those tables get overwritten.

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [x] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [x] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java [hunk 1]
```diff
@@ -110,7 +110,10 @@ public boolean canRunDataLayoutStrategyGeneration(TableMetadata tableMetadata) {
    */
   public boolean canRunDataCompaction(TableMetadata tableMetadata) {
     GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && checkCreationTimeEligibility(response) && isPrimaryTable(response);
+    return response != null
+        && checkCreationTimeEligibility(response)
+        && isPrimaryTable(response)
+        && (response.getTimePartitioning() != null || response.getClustering() != null);
```
Comment (HotSushi) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java:116]: can response.getTimePArtitioning or getClustering be empty brackets?

Comment (teamurko) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java]: we only need to check partitioning, it should be either non-empty object or null, afaiu

Comment (HotSushi) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java]: in that case should we add a check for non-empty object?

**Comment by teamurko** [comment]
obsolete

**Pull request closed without merging**